web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.test.tsx

```diff
@@ -81,3 +121,110 @@ test('runAgentAndWaitForNodeToJoin re-throws errors that are thrown while spawni
     },
   });
 });
+
+test('starting the agent flips the workspace autoStart flag to true', async () => {
+  const { appContext, rootCluster } = getMocksWithConnectMyComputerEnabled();
+  const { result } = renderHook(() => useConnectMyComputerContext(), {
+    wrapper: ({ children }) => (
+      <MockAppContextProvider appContext={appContext}>
+        <WorkspaceContextProvider value={null}>
+          <ConnectMyComputerContextProvider rootClusterUri={rootCluster.uri}>
+            {children}
+          </ConnectMyComputerContextProvider>
+        </WorkspaceContextProvider>
+      </MockAppContextProvider>
+    ),
+  });
+
+  await act(() => result.current.startAgent());
+
+  expect(
+    appContext.workspacesService.getConnectMyComputerAutoStartFlag(
+      rootCluster.uri
+    )
+  ).toBeTruthy();
+});
+
+test('killing the agent flips the workspace autoStart flag to false', async () => {
+  const { appContext, rootCluster } = getMocksWithConnectMyComputerEnabled();
+
+  const { result } = renderHook(() => useConnectMyComputerContext(), {
+    wrapper: ({ children }) => (
+      <MockAppContextProvider appContext={appContext}>
+        <WorkspaceContextProvider value={null}>
+          <ConnectMyComputerContextProvider rootClusterUri={rootCluster.uri}>
+            {children}
+          </ConnectMyComputerContextProvider>
+        </WorkspaceContextProvider>
+      </MockAppContextProvider>
+    ),
+  });
+
+  await act(() => result.current.killAgent());
```

Comment:
I remembered that the docs for `renderHook` mentioned not needing `act`, since the async utilities of that library automatically use it under the hood.

https://react-hooks-testing-library.com/usage/advanced-hooks#async

But I feel like it's better to just use `act` here as you did, because then you can have a clear test which says "When this operation finishes, expect this value to be equal to that". With `waitFor`, we'd have to wait for the attempt to finish and only then check the auto start value, which feels like a roundabout way to do it.